gateway-contracts/contracts/shared/Structs.sol

```diff
@@ -29,6 +31,26 @@ struct Coprocessor {
     string s3BucketUrl;
```

Comment:
Does it make sense to not call this s3 bucket URL, but`ciphertextUrl` or something like that? That makes it more generic, should we want to use Azure or whatever. Suggested by @antoniupop 

Btw, we could use the `featureSet` and know that from a specific feature set onwards, we start using say a different storage provider and the `ciphertextUrl` means something different.